cli/azd/cmd/middleware/hooks.go

```diff
@@ -10,99 +10,76 @@ import (
-	lazyEnv *lazy.Lazy[*environment.Environment],
-	lazyProjectConfig *lazy.Lazy[*project.ProjectConfig],
+	envManager environment.Manager,
+	env *environment.Environment,
+	projectConfig *project.ProjectConfig,
 	importManager *project.ImportManager,
 	commandRunner exec.CommandRunner,
 	console input.Console,
 	options *Options,
 ) Middleware {
 	return &HooksMiddleware{
-		lazyEnvManager:    lazyEnvManager,
-		lazyEnv:           lazyEnv,
-		lazyProjectConfig: lazyProjectConfig,
-		importManager:     importManager,
-		commandRunner:     commandRunner,
-		console:           console,
-		options:           options,
+		envManager:    envManager,
+		env:           env,
+		projectConfig: projectConfig,
+		importManager: importManager,
+		commandRunner: commandRunner,
+		console:       console,
+		options:       options,
 	}
 }
 
 // Runs the Hooks middleware
 func (m *HooksMiddleware) Run(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
-	env, err := m.lazyEnv.GetValue()
-	if err != nil {
-		log.Println("azd environment is not available, skipping all hook registrations.")
-		return next(ctx)
-	}
-
-	projectConfig, err := m.lazyProjectConfig.GetValue()
-	if err != nil || projectConfig == nil {
-		log.Println("azd project is not available, skipping all hook registrations.")
-		return next(ctx)
-	}
-
-	if err := m.registerServiceHooks(ctx, env, projectConfig); err != nil {
+	if err := m.registerServiceHooks(ctx); err != nil {
 		return nil, fmt.Errorf("failed registering service hooks, %w", err)
 	}
 
-	return m.registerCommandHooks(ctx, env, projectConfig, next)
+	return m.registerCommandHooks(ctx, next)
 }
 
 // Register command level hooks for the executing cobra command & action
 // Invokes the middleware next function
-func (m *HooksMiddleware) registerCommandHooks(
-	ctx context.Context,
-	env *environment.Environment,
-	projectConfig *project.ProjectConfig,
-	next NextFn,
-) (*actions.ActionResult, error) {
-	if projectConfig.Hooks == nil || len(projectConfig.Hooks) == 0 {
+func (m *HooksMiddleware) registerCommandHooks(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
+	if m.projectConfig.Hooks == nil || len(m.projectConfig.Hooks) == 0 {
 		log.Println(
-			"azd project is not available or does not contain any command hooks, skipping command hook registrations.",
+			"azd project does not contain any command hooks, skipping command hook registrations.",
 		)
 		return next(ctx)
 	}
 
-	envManager, err := m.lazyEnvManager.GetValue()
-	if err != nil {
-		return nil, fmt.Errorf("failed getting environment manager, %w", err)
-	}
-
-	hooksManager := ext.NewHooksManager(projectConfig.Path)
+	hooksManager := ext.NewHooksManager(m.projectConfig.Path)
 	hooksRunner := ext.NewHooksRunner(
 		hooksManager,
 		m.commandRunner,
-		envManager,
+		m.envManager,
 		m.console,
-		projectConfig.Path,
-		projectConfig.Hooks,
-		env,
+		m.projectConfig.Path,
+		m.projectConfig.Hooks,
+		m.env,
```

Comment:
If the environment isn't available and we have `nil` for environment I would expect a panic happens here.